dask/array/core.py

```diff
@@ -2750,24 +2751,30 @@ def elemwise(op, *args, **kwargs):
         # their values to compute the result dtype:
         # https://github.com/numpy/numpy/issues/6240
         # We don't inspect the values of 0d dask arrays, because these could
-        # hold potentially very expensive calculations.
-        vals = [np.empty((1,) * a.ndim, dtype=a.dtype)
+        # hold potentially very expensive calculations. Instead, we treat
+        # them just like other arrays, and if necessary cast the result of op
+        # to match.
+        vals = [np.ones((1,) * max(1, a.ndim), dtype=a.dtype)
```

Comment:
I switched to ones because I was worried about what would happen if the value was illegal for the operation e.g. 0 for log, or nan generally. But I see `apply_infer_dtype` suppresses warnings, so that's probably not an issue. I'll revert it.